src/Container/ServerRequestFactoryFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Container;
+
+use Psr\Container\ContainerInterface;
+use Zend\Diactoros\ServerRequestFactory;
+
+class ServerRequestFactoryFactory
+{
+    public function __invoke(ContainerInterface $container) : callable
+    {
+        return [ServerRequestFactory::class, 'fromGlobals'];
```

Comment:
Yes, this is a callback, but this notation is very confusing for IDEs. Not sure if PHPStorm has anything to find usages also in that notation. Maybe we should use callback as:
```php
return function () {
    return ServerRequestFactory::fromGlobals();
};
```

Reply:
That ends up being two function invocations, however. And with PHP 5.6 and up,  you can call any callable using `$callable()` notation.